src/test/database/fakeListRemote.spec.ts

```diff
@@ -0,0 +1,64 @@
+import * as pathLib from "path";
+import { expect } from "chai";
+
+import { ListRemote } from "../../database/listRemote";
+
+export class FakeListRemote implements ListRemote {
+  data: any;
+
+  /**
+   * @constructor
+   * @param data           the fake database structure. Each leaf is an integer representing the subtree's size.
+   * @param largeThreshold the threshold to determine if a delete exceeds the writeSizeLimit.
+   *                       If the sum of all leaves to delete is larger than largeThreshold,
+   *                       the delete will return false.
+   */
+  constructor(data: any) {
+    this.data = data;
+  }
+
+  listPath(path: string, numChildren: number): Promise<string[]> {
```

Comment:
Would you mind implement the `fakeRemove` and test it here?

It feels strange for the fake remote to be less capable of the real one.
```
  async listPath(
    path: string,
    numSubPath: number,
    startAfter?: string,
    timeout?: number
  ): Promise<string[]> {
```